src/FloorSubsystem/FloorSystem.java

```diff
@@ -16,18 +17,24 @@
  *
  */
 public class FloorSystem implements Runnable{
+	public final static int FLOOR_SEND_PORT = 23;
+	public final static int FLOOR_RECEIVE_PORT = 24;
+	public final static int FAULT_SEND_PORT = 25;
+	public final static int FAULT_RECEIVE_PORT = 26;
 	private FloorDataParser parser = new FloorDataParser(); // reference to the floor data parser
 	private static List<Message> floorDataEntry = new ArrayList<Message>(); // list of floor entries where each entry is a byte array
 	private ByteBufferCommunicator floorBufferCommunicator;
+	private ByteBufferCommunicator faultBufferCommunicator;
```

Comment:
Theres 2 options here. We could have 2 ByteBufferCommunicators, 1 for floor messages and 1 for faults like you have here or we could have a new ByteBufferCommunicator that handles both message types.